tests/test_process_elife_zip_dag.py

```diff
@@ -0,0 +1,85 @@
+from io import BytesIO
+from pathlib import Path
+
+import pytest
+from lxml import etree
+
+from dags.process_elife_zip_dag import (
+    extract_zipped_files_to_bucket,
+    prepare_jats_xml_for_libero,
+    ALLOWED_EXTENSIONS
+)
+from tests.assets import get_asset
+from tests.mocks import s3ClientMock
+
+
+def test_extract_zipped_files_to_bucket(mocker, context):
+    mocker.patch('boto3.client', new_callable=s3ClientMock)
+    context['dag_run'].conf = {'file': 'elife-666-vor-r1.zip'}
+    result = extract_zipped_files_to_bucket(**context)
+    assert result == 'elife-666-vor-r1/elife-666.xml'
+
+
+def test_extract_zipped_files_to_bucket_raises_exception_if_file_name_not_passed(context):
+    """
+    Should raise an exception if dag_run.conf is None.
+    """
+    dag_run = context['dag_run']
+    msg = '%s triggered without a file name passed to conf' % dag_run.dag_id
+    with pytest.raises(AssertionError) as error:
+        extract_zipped_files_to_bucket(**context)
+        assert str(error.value) == msg
+
+
+def test_extract_zipped_files_to_bucket_raises_exception_if_more_than_one_xml_in_zip(mocker, context):
+    mocker.patch('boto3.client', new_callable=s3ClientMock)
+    file_name = 'elife-666-vor-r1.zip'
+    xml_files = ['test1.xml', 'test2.xml']
+    context['dag_run'].conf = {'file': file_name}
+    namelist = mocker.patch('zipfile.ZipFile.namelist')
+    namelist.return_value = xml_files
+    msg = ('only 1 XML file supported. %s XML files found in %s: %s' %
+           (len(xml_files), file_name, xml_files))
+    with pytest.raises(AssertionError) as error:
+        extract_zipped_files_to_bucket(**context)
+        assert str(error.value) == msg
+
+
+def test_extract_zipped_files_to_bucket_raises_exception_if_no_xml_files_in_zip(mocker, context):
+    mocker.patch('boto3.client', new_callable=s3ClientMock)
+    file_name = 'elife-666-vor-r1.zip'
+    xml_files = []
+    context['dag_run'].conf = {'file': file_name}
+    namelist = mocker.patch('zipfile.ZipFile.namelist')
+    namelist.return_value = xml_files
```

Comment:
consider extracting an helper function if there is duplicated code between some tests

Reply:
refactored into a single test.